cartman.py

```diff
@@ -1,52 +1,104 @@
 import redis
 import config
+from exception import ErrorMessage
 from serializer import Serializer
 from redis_connection import Redis
+from decorators import positive_args
 
 
 class Cart(object):
 
-    """docstring for Cart"""
+    """
+        Main Class for Cart, contains all functionality
+    """
 
     def __init__(self, user_id):
-        # Init function to check weather the cart dictionary exists in redis
-        # this can also be done via user every time he uses the library
+        """
+            Constructor for the class, initializes user_id,
+            redis_connection and hash_name in redis
+        """
         self.user_id = user_id
         self.redis_connection = Redis().get_connections()
         self.redis_cart_init(user_id)
         self.destroy = self.__del__
 
     def redis_cart_init(self, user_id):
+        """
+            Get or create the hash_name in redis
+        """
         if not self.redis_connection.hexists(config.CART_CONFIG["hash_name"], self.user_id):
             self.redis_connection.hset(
                 config.CART_CONFIG["hash_name"], self.user_id, {})
 
     def add(self, product_id, unit_cost, quantity=1, **extra_data_dict):
+        """
+            Add the product of the given product_id and unit_cost with given quantity
+            Can also add extra details in the form of dictionary
+        """
         cart_dict = self.__cart_dict()
+        final_qty = quantity
+        product_id = str(product_id)
+        if cart_dict.get(product_id):
+            if cart_dict.get(product_id).get("quantity", 0) + quantity > 0:
+                final_qty = cart_dict[product_id]["quantity"] + quantity
```

Comment:
Why are you doing the developer's job. i.e. why are you not replacing the new product data entirely.
You should think it as saperation